internal/command/deploy/deploy_first.go

```diff
@@ -24,7 +25,7 @@ func (md *machineDeployment) provisionFirstDeploy(ctx context.Context) error {
 
 func (md *machineDeployment) provisionIpsOnFirstDeploy(ctx context.Context) error {
 	// Provision only if the app hasn't been deployed and have services defined
-	if !md.isFirstDeploy || len(md.appConfig.AllServices()) == 0 {
+	if !md.isFirstDeploy || len(md.appConfig.AllServices()) == 0 || flag.GetBool(ctx, "no-public-ips") {
```

Comment:
Do not do flags fetching here please, follow the pattern and pass them as deployment args at:
https://github.com/superfly/flyctl/blob/9ac4ba2da6c3a64cfd292f19f4d06aa43b255cd8/internal/command/deploy/deploy.go#L195-L205

Reply:
Yep my fault, I'll fix this now